File: tests/fixtures/ioredis/basic/cache.ts
import Redis from 'ioredis';

const redis = new Redis();
const client = new Redis();

export async function getUser(id: string) {
  return redis.get(`user:${id}`);
}

export async function setUser(id: string, name: string) {
  return redis.set(`user:${id}`, name);
}

export async function incrCounter() {
  return redis.incr('counter:requests');
}

export async function decrCounter() {
  return redis.decr('counter:errors');
}

export async function listKeys() {
  return redis.keys('user:*');
}

export async function hashSet() {
  return client.hset('profile', 'field', 'value');
}

export async function hashGetAll() {
  return client.hgetall('profile');
}

export async function setAdd() {
  return redis.sadd('active_users', 'alice');
}

export async function setRemove() {
  return redis.srem('active_users', 'alice');
}

export async function zsetAdd() {
  return redis.zadd('leaderboard', 100, 'alice');
}

export async function zsetRange() {
  return redis.zrange('leaderboard', 0, 9);
}

export async function listPush() {
  return redis.lpush('queue:jobs', 'job-data');
}

export async function listPop() {
  return redis.lpop('queue:jobs');
}

export async function expireKey() {
  return redis.expire('session:abc', 3600);
}

export async function deleteKey() {
  return redis.del('temp:abc');
}

export async function publishEvent() {
  return redis.publish('channel:events', 'event-payload');
}

export async function dynamicKey(key: string) {
  // Dynamic key — per-call-site placeholder.
  return redis.get(key);
}

File: tests/fixtures/memcache-py/basic/cache.py
"""pymemcache surface."""

from pymemcache.client.base import Client

client = Client(('localhost', 11211))


def get_user(uid: str):
    return client.get('user:1')


def set_user():
    return client.set('user:1', 'alice')


def add_entry():
    return client.add('entry:new', 'data')


def replace_entry():
    return client.replace('entry:existing', 'new-data')


def incr_counter():
    return client.incr('counter:requests', 1)


def decr_counter():
    return client.decr('counter:errors', 1)


def delete_session():
    return client.delete('session:abc')


def touch_key():
    return client.touch('session:keepalive', 60)


def get_many():
    return client.get_many(['key1', 'key2'])


def dynamic_key(k: str):
    # Dynamic — per-call-site placeholder.
    return client.get(k)
